packages/perseus/src/widgets/interactive-graphs/graphs/segment.tsx

```diff
@@ -31,19 +32,23 @@ type SegmentProps = MafsGraphProps<SegmentGraphState>;
 const SegmentGraph = ({dispatch, graphState}: SegmentProps) => {
     const {coords: segments} = graphState;
     const {strings, locale} = usePerseusI18n();
-    const lengthOfSegment =
-        segments[0] && kpoint.distanceToPoint(...segments[0]);
-
     const wholeSegmentAriaLabel = strings.srSegmentGraphAriaLabel;
-    const wholeSegmentAriaDescription =
-        segments[0] &&
-        strings.srSegmentGraphAriaDescription({
-            point1X: srFormatNumber(segments[0][0][X], locale),
-            point1Y: srFormatNumber(segments[0][0][Y], locale),
-            point2X: srFormatNumber(segments[0][1][X], locale),
-            point2Y: srFormatNumber(segments[0][1][Y], locale),
-            length: srFormatNumber(lengthOfSegment, locale),
+
+    function getLengthOfSegment(segment: PairOfPoints) {
+        return kpoint.distanceToPoint(...segment);
+    }
```

Comment:
Nit (Optional): Since this component is not using stuff inside the component (everything it needs is passed into the parameters), you can move it outside the component, maybe at the top of this file. That way it won't re-initialize on every render.

This won't really make a difference in performance in this particular case, so it's really up to you, but I believe that's considered good practice.